sklearn/linear_model/_ridge.py

```diff
@@ -1748,14 +1749,16 @@ def __init__(
     @staticmethod
     def _decomp_diag(v_prime, Q):
         # compute diagonal of the matrix: dot(Q, dot(diag(v_prime), Q^T))
-        return (v_prime * Q**2).sum(axis=-1)
+        xp, _ = get_namespace(Q)
+        return xp.sum(v_prime * Q**2, axis=1)
 
     @staticmethod
     def _diag_dot(D, B):
+        xp, _ = get_namespace(B)
```

Comment:
For my education: why is `v_prime` and `D` (respectively) not also passed to `get_namespace`? Is it worth a comment for future readers?

Reply:
they are all computed during fit so barring any bugs they should all be in the same namespace (that of `X`) but you're right that it would be clearer to pass both to `get_namespace`